lms/djangoapps/grades/transformer.py

```diff
@@ -128,6 +133,35 @@ def _collect_max_score(cls, block_structure, module):
         if max_score is None:
             log.warning("GradesTransformer: max_score is None for {}".format(module.location))
 
+    @classmethod
+    def _collect_grading_policy_hash(cls, block_structure):
+        """
+        Collect a hash of the course's grading policy, storing it as a
+        `transformer_block_field` associated with the `GradesTransformer`.
+        """
+        def _hash_grading_policy(policy):
+            """
+            Creates a hash from the course grading policy.
+            The keys are sorted in order to make the hash
+            agnostic to the ordering of the policy coming in.
+            """
+            ordered_policy = json.dumps(
+                policy,
+                separators=(',', ':'),  # Remove spaces from separators for more compact representation
+                sort_keys=True,
+            )
+            return b64encode(sha1(ordered_policy).digest())
+
+        course_key = block_structure.root_block_usage_key.course_key
+        course_descriptor = get_course_by_id(course_key, depth=0)
```

Comment:
Rather than getting the course from the modulestore again, replace the above 2 lines with:
```
course_location = block_structure.root_block_usage_key
block_structure.get_xblock(course_location)
```